oneview/data_source_interconnects.go

```diff
@@ -0,0 +1,1226 @@
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"scopes_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"serial_number": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"snmp_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"category": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"created": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"eTag": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"modified": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"read_community": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"state": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"system_contact": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"v3_enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"spare_part_number": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"stacking_domain_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"stacking_domain_role": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"stacking_member_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"storm_control_polling_interval": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"storm_control_threshold": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"sub_port_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"uid_state": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"unsupported_capabilities": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func dataSourceInterconnectsRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	name := d.Get("name").(string)
```

Comment:
Is it only the name with which we can get the resource or it can be done using URI as well?

Reply:
Can be done with uri, but data source should be through name right?